src/main/java/org/spongepowered/asm/mixin/transformer/ClassInfo.java

```diff
@@ -2025,9 +2080,13 @@ public static ClassInfo forName(String className) {
         ClassInfo info = ClassInfo.cache.get(className);
         if (info == null) {
             try {
-                int flags = MixinEnvironment.getCurrentEnvironment().getOption(Option.CLASSREADER_EXPAND_FRAMES) ? ClassReader.EXPAND_FRAMES : 0;
-                ClassNode classNode = MixinService.getService().getBytecodeProvider().getClassNode(className, true, flags);
-                info = new ClassInfo(classNode);
+                if (className.startsWith("java/")) { // this would ideally check the platform class loader for other jdk classes, but needs extra api to do so
```

Comment:
Actually, why not use `ClassLoader.getPlatformClassLoader().getResource(...)` to check instead of this?

Reply:
Not an option on [Java 8](https://docs.oracle.com/javase/9/docs/api/java/lang/ClassLoader.html#getPlatformClassLoader--)

Reply:
We know our ASM supports Java 8 so we can skip the check there.